include/cpr/proxies.h

```diff
@@ -13,6 +13,7 @@ class Proxies {
     Proxies() {}
     Proxies(const std::initializer_list<std::pair<const std::string, std::string>>& hosts);
 
+    void add(const std::string& protocol, std::string proxy_url);
```

Comment:
`std::string proxy_url` should be `std::string&& proxy_url`, so the user knows about the copy of `proxy_url`.

Reply:
> Yes, there is line in `session.cpp`:
> 
> ```c++
> ...
> curl_easy_setopt(curl, CURLOPT_PROXY, proxies_[protocol].c_str());
> ...
> ```
> 
> `c_str()` is a const function so it cannot be called on a non-const object. This makes my original change unviable. Now I've just added and extra method to the Proxy class for adding a new proxy.

It's totally fine to call a const function on a non-const object.  It looks fine to me to remove the constness and make it modifiable like Cookies is.  Proxies is read once by Session at the start of every request, and the Session object has its own storage, so user modification to the object after it is passed would have no effect anyway.

Reply:
i was looking for something similar to this pattern to understand it better and found https://github.com/whoshuu/cpr/blob/master/include/cpr/curl_container.h#L14 which is used by https://github.com/whoshuu/cpr/blob/master/include/cpr/curl_container.h#L42 for basically the same thing in the Parameters object.  `add` is capitalized to `Add` and both `const &` and `&&` forms are provided.